turbopack/crates/turbo-persistence/src/static_sorted_file_builder.rs

```diff
@@ -94,387 +94,472 @@
     pub entries: u64,
 }
 
-#[derive(Debug, Default)]
-pub struct StaticSortedFileBuilder<'a> {
-    aqmf: Cow<'a, [u8]>,
-    key_compression_dictionary: Vec<u8>,
-    value_compression_dictionary: Vec<u8>,
-    blocks: Vec<(u32, Vec<u8>)>,
-    min_hash: u64,
-    max_hash: u64,
-    entries: u64,
-}
-
-impl<'a> StaticSortedFileBuilder<'a> {
-    pub fn new<E: Entry>(
-        entries: &[E],
-        total_key_size: usize,
-        total_value_size: usize,
-    ) -> Result<Self> {
-        debug_assert!(entries.iter().map(|e| e.key_hash()).is_sorted());
-        let mut builder = Self {
-            min_hash: entries.first().map(|e| e.key_hash()).unwrap_or(u64::MAX),
-            max_hash: entries.last().map(|e| e.key_hash()).unwrap_or(0),
-            entries: entries.len() as u64,
-            ..Default::default()
-        };
-        builder.compute_aqmf(entries);
-        builder.compute_compression_dictionary(entries, total_key_size, total_value_size)?;
-        builder.compute_blocks(entries);
-        Ok(builder)
+pub fn write_static_stored_file<E: Entry>(
+    entries: &[E],
+    total_key_size: usize,
+    total_value_size: usize,
+    file: &Path,
+) -> Result<(StaticSortedFileBuilderMeta<'static>, File)> {
+    debug_assert!(entries.iter().map(|e| e.key_hash()).is_sorted());
+
+    let mut file = BufWriter::new(File::create(file)?);
+
+    // We use a shared buffer for all operations to avoid excessive allocations
+    let mut buffer = Vec::new();
+
+    reserve_compression_buffer(&mut buffer, total_key_size, total_value_size);
```

Comment:
nit: just have this allocate the Vec with `Vec::with_capacity` instead of using `reserve`